src/crl/interactivesessions/shells/rawpythonshell.py

```diff
@@ -45,7 +45,14 @@ def _verify_setup(self):
 
     def _read_until_prompt_after_last_command(self):
         last = self.setup_cmds[-1]
-        out = self._read_until(last, timeout=self.short_timeout)
+        out = ''
+        for _ in range(int(float(self.short_timeout) / float(0.1))):
+            try:
+                out += self._read_until(last, 0.1)
+                break
+            except TimeoutError:
+                self._sendline(last)
+                continue
         out += self._read_until_prompt(timeout=self.short_timeout)
```

Comment:
it is raised here, likely, but still the correct location would be to raise if we do 'endofsetup'.